src/main/java/duke/DoneInput.java

```diff
@@ -0,0 +1,24 @@
+package duke;
+
+public class DoneInput extends InputHandler {
+
+    public DoneInput(Ui ui, TaskList taskList) throws DukeException {
+        super(ui, taskList);
+    }
+
+    @Override
+    public String handle (String input) throws EmptyDescriptionException {
+        if (input.length() == 4) {
+            throw new EmptyDescriptionException("error");
```

Comment:
I think it's great to have different Exceptions here! really improves the OO-ness of the project.